### PR TITLE
Fix issue with x, y axes being cut off with long labels

### PR DESCRIPTION
**Pull request by miguelpdiaz8**
## Description

Fix issue with x, y axes being cut off with long labels

## Testing

Tested it in my local environment. You can test it here: https://migdizn.badger.wwps.aws.dev/admin/dashboard/ab13ccfa-17e5-4f5f-826d-0c628a621edc/edit-chart/f3b20700-fa48-4c54-8004-43b675dbd071

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: frontend/src/components/BarChartPreview.tsx [hunk 1]
```diff
@@ -86,18 +90,13 @@ const BarChartPreview = (props: Props) => {
...
-                  )
-                  .map((c) => (c as string).length)
-                  .reduce((a, b) => (a > b ? a : b), 0) || 0) *
-                  8 +
-                24
-              }
+              width={Math.min(
+                UtilsService.getLargestHeader(props.bars, props.data) * 8 + 24,
+                220
+              )}
```
Comment (ferdingler) [inline frontend/src/components/BarChartPreview.tsx]: It feels like lately we have been adding a lot of hardcoded "magical" numbers on the Chart components. Is there any way we can make this better?

File: frontend/src/components/ColumnChartPreview.tsx [hunk 2]
```diff
@@ -50,8 +52,15 @@ const ColumnChartPreview = (props: Props) => {
     }
   };
 
+  const widthPercent =
+    (UtilsService.getLargestHeader(columns, data) *
+      (data ? data.length : 0) *
+      8 *
+      100) /
+    (props.isPreview ? 480 : 960);
```
Comment (ferdingler) [inline frontend/src/components/ColumnChartPreview.tsx]: Similar comment. `480`, `960`, `8 * 100`, `0`. What do these numbers mean? Are we going to remember in the future? What about making them constants on the top of the file to give them descriptive names at least? Would that help?

File: frontend/src/components/BarChartPreview.tsx (hunk 1 above)
Comment (miguelpdiaz8) [reply frontend/src/components/BarChartPreview.tsx]: Yeah, I agree, those are kind of magic numbers, I had to spend a lot of time figuring that out, and I dont think it is perfect yet. The problem is recharts does not support some functionalities, so we need to do the calculations manually. One way I see we could do it better is to create variables for each number describing what they represent (offset, pixel by character, etc)

File: frontend/src/components/ColumnChartPreview.tsx (hunk 2 above)
Comment (miguelpdiaz8) [reply frontend/src/components/ColumnChartPreview.tsx]: Same

**Review by ferdingler — APPROVED** [review]

**Pull request merged**